storage/federatedx/ha_federatedx.cc

```diff
@@ -1511,7 +1511,7 @@ static void fill_server(MEM_ROOT *mem_root, FEDERATEDX_SERVER *server,
   server->key= (uchar *)  strmake_root(mem_root, key.ptr(), key.length());
 
   /* pointer magic */
-  server->scheme+= (intptr) server->key;
+  server->scheme= (const char *)server->key;
   server->hostname+= (intptr) server->key;
   server->database+= (intptr) server->key;
   server->username+= (intptr) server->key;
```

Comment:
let's fix this "pointer magic"? I don't think it helps much and makes the code difficult to follow. Something like
```diff
-  server->hostname= (const char *) (intptr) key.length();
+  size_t hostname_pos= key.length();
```
and
```diff
-  server->hostname+= (intptr) server->key;
+  server->hostname= server->key + hostname_pos;
```
and, of course
```diff
- /* pointer magic */
```
☺